File: lang/go/src/fun/vac/algorithms/number_theory/LeastCommonMultiple/LeastCommonMultiple.go
package LeastCommonMultiple

import "fun/vac/algorithms/number_theory/GreatestCommonDivisor"

func ReduceToBinaryGCD(m int64, n int64) int64 {
    if m < 0 {
        m = -m
    }
    if n < 0 {
        n = -n
    }

    if m == 0 || n == 0 {
        return 0
    }
    return m / GreatestCommonDivisor.IterativeBinaryGCD(m, n) * n
}

func ReduceToEuclidean(m int64, n int64) int64 {
    if m < 0 {
        m = -m
    }
    if n < 0 {
        n = -n
    }

    if m == 0 || n == 0 {
        return 0
    }
    return m / GreatestCommonDivisor.IterativeEuclidean(m, n) * n
}

File: archive/nodejs/muse/algorithms/number_theory/LeastCommonMultiple.js
import * as GreatestCommonDivisor from './GreatestCommonDivisor.js';

export const reduceToBinaryGCD = (m, n) => {
  if (m < 0) {
    m = -m;
  }
  if (n < 0) {
    n = -n;
  }

  if (m === 0 || n === 0) {
    return 0;
  }
  return (m / GreatestCommonDivisor.iterativeBinaryGCD(m, n)) * n;
};

export const reduceToEuclidean = (m, n) => {
  if (m < 0) {
    m = -m;
  }
  if (n < 0) {
    n = -n;
  }

  if (m === 0 || n === 0) {
    return 0;
  }
  return (m / GreatestCommonDivisor.iterativeEuclidean(m, n)) * n;
};
